dnf/cli/cli.py

```diff
@@ -214,13 +214,12 @@ def do_transaction(self, display=()):
             elif 'test' in self.conf.tsflags:
                 logger.info(_("{prog} will only download packages, install gpg keys, and check the "
                               "transaction.").format(prog=dnf.util.MAIN_PROG_UPPER))
-            if dnf.util.is_container():
-                _container_msg = _("""
-*** This system is managed with ostree.  Changes to the system
-*** made with dnf will be lost with the next ostree-based update.
-*** If you do not want to lose these changes, use 'rpm-ostree'.
+            if dnf.util.is_bootc_host():
+                _bootc_host_msg = _("""
+*** error: system is configured to be read-only; for more
+*** information run `bootc status` or `ostree admin status`.
```

Comment:
Definitely tempting to mention `bootc usroverlay` here, but probably better for now to leave that for when we link to a docs page.

Reply:
Yeah..., also I am not sure if we should mention ostree admin status and just mention bootc status. Idk WDYT @cgwalters ?

something like:
```
*** error: system is configured to be read-only; for more
*** information run `bootc status`. To make transient
*** changes (removed after a reboot) to your system
*** use `bootc usroverlay` before running dnf.
```

Reply:
Also, I re-thinking the link in the error message, I am not sure if that is going to be great to maintain. If the page goes 404, and we update it, there is a chance people will still get a 404. I would be happier with a verbose message instead.

Reply:
Well, at least for enterprise downstream OS/distros I am pretty sure we can provide a stable link. I am not sure if this topic has come up before in dnf at all, but I could imagine the URL being a build-time option.

The problem with embedding messages in the code is that I am pretty sure (at least hopeful) that the recommendations will evolve faster than we want to rev the code specifically.

Reply:
> Well, at least for enterprise downstream OS/distros I am pretty sure we can provide a stable link. I am not sure if this topic has come up before in dnf at all, but I could imagine the URL being a build-time option.

AFAIK we don't have those. (But i am not sure) But references to commands is good choice. We spoke about this in team and maybe references to man pages which can have specific links would be also a good option.

Reply:
I agree but I don't really want to hold up this commit to hash out the details of this message.  That can be a separate cleanup.

Reply:
Alright, I see "bootc status" and "ostree admin status" in the message which I deem good enough for this PR.

Reply:
I don't quite understand the reference to `bootc status`, which today is...not itself directly obviously helpful for people unfamiliar with the system.

We could obviously *make* `bootc status` have this role, but maybe we need `bootc help` or I guess we could just put more information in `bootc --help` which already exists.

Yeah, I like that one.

Then indeed we can have a link to a URL in `bootc --help` that's per distro/OS.

Reply:
I'd personally also prefer that if bootc is present, we don't *also* tell people about `ostree admin status` as that's less maintained/useful.

Reply:
since we are really adding dnf to systems that already have bootc, I think we could just always just mention bootc.

What about a simple:

```
*** Error: system is configured to be read-only; for more
*** information run `bootc --help`.
```

Reply:
Let's continue on: https://github.com/rpm-software-management/dnf/pull/2112